staging/src/k8s.io/apimachinery/pkg/util/managedfields/internal/structuredmerge.go

```diff
@@ -71,6 +74,7 @@ func NewCRDStructuredMergeManager(typeConverter TypeConverter, objectConverter r
 			Converter:     newCRDVersionConverter(typeConverter, objectConverter, hub),
 			IgnoredFields: resetFields,
 		},
+		allowDuplicatesInSets: false,
```

Comment:
CRs can have duplicates in sets if they were persisted prior to the schema being updated to mark a list a set, should this be true? should we have a test?

Reply:
Yeah that's a good point, we should probably allow it. The test won't add much though? it's basically the exact same path as built-ins.

Reply:
ok, I didn't know if there were different paths taken for typed/structured vs untyped/openapi schema only

Reply:
I've updated it, it simplifies the code, I like it :-)